src/main/java/core/ClientHandler.java

```diff
@@ -85,6 +98,14 @@ public JsonObject getClientConnectionDataAsJson() {
         return clientRecord;
     }
 
+    private boolean isClientConnected() throws IOException {
+        return this.input.read() != -1;
```

Comment:
I would add try catch for IOException and return false when the exception is called.